### PR TITLE
7.x binary tests

### PR DESCRIPTION
**Pull request by qadan**
The derivatives test contains the actual binary tests.

### CODE REVIEW

File: tests/islandora_book_ingest_purge.test [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    $book_object = $this->getObjectFromPath($this->url);
+    $datastreams = array(
+      'RELS-EXT',
+      'DC',
+    );
+    $this->assertDatastreams($book_object, $datastreams);
+
+    // Attempts to add a page to the book.
+    $path = 'islandora/object/' . $book_object . '/manage/book/ingest';
+    $file = drupal_get_path('module', 'islandora_book') . '/tests/fixtures/test.tiff';
```
Comment (jordandukart) [inline tests/islandora_book_ingest_purge.test:81]: Missing fixtures folder in commit.

File: tests/islandora_book_derivatives_ingest_purge.test [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    $book_object = $this->getObjectFromPath($this->url);
+    $datastreams = array(
+      'RELS-EXT',
+      'DC',
+    );
+    $this->assertDatastreams($book_object, $datastreams);
+
+    // Attempts to add a page to the book.
+    $path = 'islandora/object/' . $book_object . '/manage/book/ingest';
+    $file = drupal_get_path('module', 'islandora_book') . '/tests/fixtures/test.tiff';
```
Comment (jordandukart) [inline tests/islandora_book_derivatives_ingest_purge.test:104]: Missing fixtures folder in commit.

File: tests/islandora_book_derivatives_ingest_purge.test [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+      'HOCR',
+    );
+    $this->assertDatastreams($page_object, $datastreams);
+
+    // Attempts to verify basic image datastreams.
+    $datastreams = array(
+      'JPG',
+      'TN',
+    );
+    $this->assertImageDatastreams($page_object, $datastreams);
```
Comment (jordandukart) [inline tests/islandora_book_derivatives_ingest_purge.test:135]: `assertImageDatastreams` is not defined.

File: tests/islandora_book_derivatives_ingest_purge.test [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+    // Attempts to match more complex image signatures to expected values.
+    $datastreams = array(
+      array('JP2', IMAGETYPE_JP2),
+      array('OBJ', IMAGETYPE_TIFF_II),
+    );
+    $this->assertImageDatastreamType($page_object, $datastreams);
```
Comment (jordandukart) [inline tests/islandora_book_derivatives_ingest_purge.test]: `assertImageDatastreamType` is not defined.

**Comment by qadan** [comment]
this is like the grossest commit i ever did done

**Comment by christianselig** [comment]
You don't deserve an Ubuntu Edge.

File: tests/islandora_book_ingest_purge.test (hunk 1 above)
Comment (qadan) [inline tests/islandora_book_ingest_purge.test:81]: Exists now! mmm hmm.

File: tests/islandora_book_derivatives_ingest_purge.test (hunk 3 above)
Comment (qadan) [inline tests/islandora_book_derivatives_ingest_purge.test:135]: Is defined now! Yup yup.

**Pull request closed without merging**